### ISSUE TITLE
Cannot find module tsconfig-paths\lib\tsconfig-loader in ESM mode

### ISSUE THREAD

**Issue by gunters63** [aleclarson/vite-tsconfig-paths#27]
I want to switch to ESM mode (`"type": "module"`).
vite-tsconfig-paths gives me above error.
It seems ESM should be supported because I can see 
````
  "exports": {
    "import": "./dist/index.mjs",
    "require": "./dist/index.js"
  },
````
in the package.json of vite-tsconfig-paths so it should be a hybrid module.

When I look into dist/index.mjs there is the following include:
````
import {
  loadTsconfig,
  walkForTsConfig
} from "tsconfig-paths/lib/tsconfig-loader";
````

which fails

**Comment by gunters63** [comment]
Sorry for the nonsense, tsconfig-paths is an external module, I should put it in my package.json too ( I am using pnpm)

**Comment by gunters63** [comment]
Strangely it works when I patch index.mjs like this:

````
import {
  loadTsconfig,
  walkForTsConfig
} from "tsconfig-paths/lib/tsconfig-loader.js";
````

I have no idea why he cannot resolve the module without the explicit extension

**Comment by aleclarson** [comment]
Whoops. The explicit `.js` extension was included in #24, but I naïvely removed it without asking why it was added. 😅 

[This page](https://github.com/nodejs/node/blob/master/doc/api/esm.md#import-specifiers) describes why it's necessary (see the last sentence below):

<img width="633" alt="image" src="https://user-images.githubusercontent.com/1925840/119847103-b1322900-bed8-11eb-869d-d7ec45fd9f40.png">

**Comment by aleclarson** [comment]
Released in v3.3.13
